rust/stackable-cockpitd/src/api_doc.rs

```diff
@@ -57,7 +57,7 @@ pub fn openapi() -> utoipa::openapi::OpenApi {
 }
 
 /// Synthetic types that are used to generate type definitions for foreign types.
-mod synthetic_types {
+pub mod synthetic_types {
```

Comment:
I assume this is ok. Otherwise we can do:

```suggestion
#[allow(dead_code)]
mod synthetic_types {
```

Reply:
This _should_ be fine, as this is not a library and as such doesn't expose any of the types (and modules in this case).

I still think that this should be kept private, because it is only used during OpenAPI spec generation (and I thus consider it as an internal implementation detail). So I would prefer to add `#[allow(dead_code)]` here. But let's also add a comment for what the module is used and why we added the attribute.

Reply:
There is already a comment (it's just further below, so I'll move it up).
I'll keep it private, and use the `allow` instead.

Reply:
Could we maybe extend the comment and state why the attribute was needed. For the future reader it might be immediately obvious why it's there.

Reply:
Done in 577b4534e9b48b81db1918b9c31ebe894fd79183